Spam-protected relay protocol raw spec

staheri14 opened the pull request:
Addresses the following issues https://github.com/vacp2p/specs/issues/242 , https://github.com/vacp2p/specs/issues/239, https://github.com/vacp2p/specs/issues/240, and https://github.com/vacp2p/research/issues/49 will be improved in the follow up PRs.

staheri14 committed:
WIP: adds the draft of spam-protected relay protocol


minor edits


adds minor edits


fixes minor typos

staheri14 committed:
updates wordlist

staheri14 committed:
adds some headers to the flow description

staheri14 committed:
adds the revised version

staheri14 committed:
Update waku-spam-protected-relay.md

staheri14 committed:
Delete waku-spam-protection.md

staheri14 committed:
Auto stash before merge of "spam-protection/solution-overview" and "origin/spam-protection/solution-overview"

staheri14 committed:
minor

staheri14 committed:
Update waku-spam-protected-relay.md

staheri14 committed:
minor

staheri14 committed:
Merge remote-tracking branch 'origin/spam-protection/solution-overview' into spam-protection/solution-overview

oskarth commented on specs/waku/v2/waku-spam-protected-relay.md:
```diff
@@ -0,0 +1,130 @@
+---
+title: Waku Spam Protected Relay
```
I'd call this something like Waku RLN Relay or to that effect, otherwise seems like the existing relay isn't spam protected at all. Plus this opens up door for other spam protection mechanisms.

staheri14 replied:
Thanks! your suggested title makes so much sense, will apply the change.

oskarth commented:
Didn't get to give this a thorough read today, will have a closer look tomorrow!

oskarth reviewed (approved):
Looks great as a raw spec! Discussed during call a bit

oskarth commented on specs/waku/v2/waku-spam-protected-relay.md:
```diff
@@ -0,0 +1,130 @@
+-------
+
+# Protobuf
+
```
Discussed during call, add note for WakuMessage upstream spec

oskarth commented on specs/waku/v2/waku-spam-protected-relay.md:
```diff
@@ -0,0 +1,130 @@
+
+TODO: may consider [validator functions](https://github.com/libp2p/specs/tree/master/pubsub#topic-validation) or [extended validators](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#extended-validators) for the spam detection
+
+# Security Considerations
```
Discussed during call, not for this PR but later enhancement - economic spam guarantees

jm-clius commented on specs/waku/v2/waku-spam-protected-relay.md:
```diff
@@ -0,0 +1,130 @@
+  optional bytes payload = 1;
+  optional fixed32 contentTopic = 2;
+  optional uint32 version = 3;
+  optional ProofBundle proofBundle = 4;
```
I think in proto3 `optional` is omitted as in `ProofBundle` below.

staheri14 replied:
I agree, I think after the removal of `required` in proto3 the `optional` declaration is also redundant , but I wanted to be consistent with the general style of other waku specs :)

staheri14 replied:
Indeed, I borrowed the WakuMessage protobuf from its own spec, that is why `optional` is still there

oskarth replied:
Yeah we should probably clean that up. It is worth noting that IIRC libp2p uses a mix of proto2 and proto3, and aren't always explicit about it

jm-clius reviewed (approved):
LGTM! 👍

staheri14 committed:
updates wordlist


adds more TODOs


adds root


adds field numbers


adds the security consideration section


renames waku_rln_relay


adds note for WakuMessage protobuf


adds more TODOs
